### PR TITLE
feat: add `upgrade` cli command

### PR DESCRIPTION
**Pull request by msga-mmm**
Add command to upgrade `whiz` to the latest version from the binaries released to GitHub.

The binaries are released by a workflow that is triggered when a new tag is created.

This fixes issue #20.

### CODE REVIEW

File: .github/workflows/release-artifacts.yml [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+          a \
+          -mm=Deflate \
+          -mfb=273 \
+          -mpass=15 \
+          "${{ env.artifact_name }}" \
+          "${{ env.build_name }}"
+
+      - uses: ncipollo/release-action@v1
+        with:
+          allowUpdates: true
```
Comment (zifeo) [inline .github/workflows/release-artifacts.yml:69]: add an bump commit like in meta?

File: src/main.rs [hunk 2]
```diff
@@ -14,6 +15,9 @@ use clap::Parser;
 #[derive(Parser, Debug)]
 #[clap(author, version, about, long_about = None)]
 struct Args {
+    #[clap(subcommand)]
+    command: Option<Command>,
```
Comment (zifeo) [inline src/main.rs:19]: can we make it `whiz self upgrade` so we can simplify single job run like [here](https://github.com/zifeo/whiz/issues/27)

**Comment by zifeo** [comment]
1. let’s try to make it as close as https://github.com/metatypedev/metatype/blob/main/.github/workflows/release.yml in format, file and naming

2. I believe it should be restrict to github tag https://github.com/zifeo/whiz/pull/26/files#diff-89712d434a9475cf0d1a506d41228294e6d4312a695f2dbfa56daabb1a2418bbR6

3. I believe this is already the default? https://github.com/zifeo/whiz/pull/26/files#diff-89712d434a9475cf0d1a506d41228294e6d4312a695f2dbfa56daabb1a2418bbR28

4. use Swatinem/rust-cache@v2 for cache https://github.com/zifeo/whiz/pull/26/files#diff-89712d434a9475cf0d1a506d41228294e6d4312a695f2dbfa56daabb1a2418bbR33

5. you may skip the env var (path, build?) and directly re-use the matrix variable in the next step https://github.com/zifeo/whiz/pull/26/files#diff-89712d434a9475cf0d1a506d41228294e6d4312a695f2dbfa56daabb1a2418bbR50 (edited) 

6. backport into meta-cli ? https://github.com/zifeo/whiz/pull/26/files#diff-2e9d962a08321605940b5a657135052fbcef87b5e360662bb527c96d9a615542R31

**Comment by msga-mmm** [comment]
> 1. let’s try to make it as close as https://github.com/metatypedev/metatype/blob/main/.github/workflows/release.yml in format, file and naming
> 
> 2. I believe it should be restrict to github tag https://github.com/zifeo/whiz/pull/26/files#diff-89712d434a9475cf0d1a506d41228294e6d4312a695f2dbfa56daabb1a2418bbR6
>
> 3. I believe this is already the default? https://github.com/zifeo/whiz/pull/26/files#diff-89712d434a9475cf0d1a506d41228294e6d4312a695f2dbfa56daabb1a2418bbR28
>
> 4. use Swatinem/rust-cache@v2 for cache https://github.com/zifeo/whiz/pull/26/files#diff-89712d434a9475cf0d1a506d41228294e6d4312a695f2dbfa56daabb1a2418bbR33
>
> 5. you may skip the env var (path, build?) and directly re-use the matrix variable in the next step https://github.com/zifeo/whiz/pull/26/files#diff-89712d434a9475cf0d1a506d41228294e6d4312a695f2dbfa56daabb1a2418bbR50 (edited)
> 
> 6. backport into meta-cli ? https://github.com/zifeo/whiz/pull/26/files#diff-2e9d962a08321605940b5a657135052fbcef87b5e360662bb527c96d9a615542R31

All the suggestions were implemented, thank you for the feedback.

**Review by zifeo — APPROVED** [review]

**Pull request merged**